code/modules/research/xenobiology/crossbreeding/warping

```diff
@@ -0,0 +1,796 @@
+
+/obj/effect/warped_rune/purplespace
+	desc = "When all that was left were plastic walls and the clothes on their back, they knew what they had to do."
+	icon_state = "purplespace"
+	var/obj/item/stack/medical/suture/S          // S stands for suture because it just make sense
+	var/obj/item/stack/medical/mesh/M            //M is for mesh and you
+	max_cooldown = 30
+
+/obj/effect/warped_rune/purplespace/Initialize()
+	.=..()
+	START_PROCESSING(SSprocessing, src)
+
+
+/obj/effect/warped_rune/purplespace/process()
+	if(cooldown < world.time)
+		cooldown = world.time + max_cooldown
+		transmute_heal()
+
+
+///""transforms"" cloth and plastic into suture and regenerative mesh
+/obj/effect/warped_rune/purplespace/proc/transmute_heal() //transmute plastic and cloth into suture and regenerative mesh
+	for(var/obj/item/stack/sheet/plastic/P in T)  //transmute Plastic into regenerative mesh
+		M = new (T,1)
+		P.amount--
+		playsound(T, 'sound/effects/splat.ogg', 20, TRUE)
+		if(P.amount <= 0)
+			qdel(P)
+
+	for(var/obj/item/stack/sheet/cloth/C in T) //transmute cloth into suture
+		S = new(T, 1)
+		C.amount--
+		playsound(T, 'sound/effects/splat.ogg', 20, TRUE)
+		if(C.amount <= 0)
+			qdel(C)
+
+
+
+
+
+
+/* the blue warp rune  keeps a tile slippery CONSTANTLY by adding water (maybe I'll make it lube if I feel extra fucky) every few(10) seconds */
+
+
+
+/obj/item/slimecross/warping/blue
+	colour = "blue"
+	runepath = /obj/effect/warped_rune/cyanspace //we'll call the blue rune cyanspace to not mix it up with actual bluespace rune
+	effect_desc = "creates a rune that constantly wet itself with water as long as the rune is up"
+
+/obj/effect/warped_rune/cyanspace
+	icon_state = "slipperyspace"
+	desc = "Water drowns, water feeds, water slips, is there a thing water cannot do?"
+
+/obj/effect/warped_rune/cyanspace/Initialize()
+	.=..()
+	START_PROCESSING(SSprocessing, src)
+
+/obj/effect/warped_rune/cyanspace/process()
+	if(cooldown < world.time)
+		cooldown = world.time + max_cooldown
+		slippery_rune(T)
+
+
+///just wet the turf under the rune every now and then, there's actually a small moment where the tile isn't wet
+/obj/effect/warped_rune/cyanspace/proc/slippery_rune(turf/open/T)
+	T.MakeSlippery(TURF_WET_WATER, min_wet_time = 10 SECONDS, wet_time_to_add = 5 SECONDS)
+
+
+
+
+
+/*      Metal rune : makes an invisible wall. actually I lied, the rune is the wall.*/
+
+
+/obj/item/slimecross/warping/metal
+	colour = "metal"
+	runepath = /obj/effect/warped_rune/metalspace
+	effect_desc = "Creates a rune that prevents passage above it, takes longer to store and draw than other runes."
+	storing_time = 200  //storing time is longer for this particular rune
+	drawing_time = 100  //it's also longer to draw it
+
+//It's a wall what do you want from me
+/obj/effect/warped_rune/metalspace
+	desc = "Words are powerful things, they can stop someone dead in their tracks if used in the right way"
```

Comment:
need a period at the end of this

also, I'm not sure if that comma should be replaced with another kind of punctuation mark. commas were never really my strong point, so I think I'm gonna mostly ignore them from here on out in this review, since I don't _really_ know what I'm talking about as well as I should when it comes to commas.